velox/functions/sparksql/URLFunctions.h

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <boost/regex.hpp>
+#include <folly/CPortability.h>
+#include <cctype>
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+/// Performs initial validation of the URI.
```

Comment:
/// for public APIs, // for private

Reply:
Resolved. In this case, would that mean the comment above ParseUrlFunction should be /// ? As of now it is 

```cpp
// parse_url(url, partToExtract) → string
// parse_url(url, partToExtract, key) → string
//
// Extracts a part of a URL. The partToExtract argument can be one of
// 'PROTOCOL', 'HOST', 'PATH', 'REF', 'AUTHORITY', 'FILE', 'USERINFO',
// 'QUERY', 'FRAGMENT'.
//
// If the URL is invalid, return null.
template <typename T>
struct ParseUrlFunction {
```